scripts/sfpshow

```diff
@@ -490,16 +499,56 @@ class SFPShow(object):
     def beautify_pm_field(self, prefix, field):
         if field is None:
             return 'N/A'
-        elif prefix in {'prefec_ber'}:
+        elif prefix in {'prefec_ber'} and isinstance(field, float):
             return "{:.2E}".format(field) if field != 0 else '0.0'
         else:
             return str(field)
 
-    def convert_interface_sfp_pm_to_cli_output_string(self, state_db, interface_name):
-        sfp_pm_dict = state_db.get_all(
-            self.db.STATE_DB, 'TRANSCEIVER_PM|{}'.format(interface_name))
-        sfp_threshold_dict = state_db.get_all(
-            state_db.STATE_DB, 'TRANSCEIVER_DOM_THRESHOLD|{}'.format(interface_name))
+    def beautify_pm_timestamp(self, secs_since_epoch):
+        if secs_since_epoch is None:
+            return 'N/A'
+        dt_object = datetime.utcfromtimestamp(secs_since_epoch)
+        formatted_time = dt_object.strftime("%a %b %d %H:%M:%S UTC %Y")
+        return formatted_time
+
+    def get_pm_dict_for_specified_window(self, db, interface_name,
+                                         window_size, window_index):
+        """Get PM dict for specified window size and index"""
+        pm_windows_dict = db.get_all(
+            db.STATE_DB, '{}|{}'.format(self.PM_TABLE_NAME_IN_DB, interface_name))
+        if not pm_windows_dict:
+            return None
+        window_num = self.PM_WINDOW_NUM_IN_DB[window_size]
+        start_win_idx_in_db = self.PM_WINDOW_START_INDEX_IN_DB[window_size]
+        end_win_idx_in_db = start_win_idx_in_db + window_num - 1
+        pm_windows_for_win_size = []
+        cur_win_idx = -1
+        for i in range(start_win_idx_in_db, end_win_idx_in_db + 1):
+            win_name = 'window' + str(i)
+            if (win_name not in pm_windows_dict or
+                pm_windows_dict[win_name] == 'N/A'):
+                pm_dict = None
+            else:
+                # Convert string to dict
+                pm_dict = ast.literal_eval(pm_windows_dict[win_name])
+            pm_windows_for_win_size.append(pm_dict)
+            if (pm_dict is not None and
+                'pm_win_current' in pm_dict and
+                pm_dict['pm_win_current'] == 'true'):
+                cur_win_idx = i
+        if cur_win_idx == -1:
+            return None
+        return pm_windows_for_win_size[(cur_win_idx -
+                                        start_win_idx_in_db -
+                                        window_index +
+                                        window_num) % window_num]
+
+    def convert_interface_sfp_pm_to_cli_output_string(self, db, interface_name,
+                                                      window_size, window_index):
+        sfp_pm_dict = self.get_pm_dict_for_specified_window(
```

Comment:
if requested history window is not completed/filled, then display a message instead of empty o/p. 'PM data not available for the requested window'

Reply:
updated it as such